src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/PartialUpdateSpec.groovy

```diff
@@ -529,6 +551,92 @@ class PartialUpdateSpec extends HealthCheckSpecification {
         ]
     }
 
+    @Tidy
+    @Tags(HARDWARE)
+    def "Able to update flow encapsulationType using partial update"() {
+        given: "A flow with a 'transit_vlan' encapsulation"
+        def switchPair = topologyHelper.getAllNeighboringSwitchPairs().find {
+            it.src.noviflow && !it.src.wb5164 && it.dst.noviflow && !it.dst.wb5164
+        }
+        assumeTrue("Unable to find required switches in topology", switchPair as boolean)
+
+        def flow = flowHelperV2.randomFlow(switchPair)
+        flow.encapsulationType = FlowEncapsulationType.TRANSIT_VLAN
+        flowHelperV2.addFlow(flow)
+
+        def originalCookies = northbound.getSwitchRules(switchPair.src.dpId).flowEntries.findAll {
+            !new Cookie(it.cookie).serviceFlag
+        }
+
+        when: "Request a flow partial update for an encapsulationType field(vxlan)"
+        def newEncapsulationTypeValue = FlowEncapsulationType.VXLAN.toString().toLowerCase()
+        def updateRequest = new FlowPatchV2().tap { it.encapsulationType = newEncapsulationTypeValue }
+        def response = flowHelperV2.partialUpdate(flow.flowId, updateRequest)
+
+        then: "Update response reflects the changes"
+        response.encapsulationType == newEncapsulationTypeValue
+
+        and: "Changes actually took place"
+        northboundV2.getFlow(flow.flowId).encapsulationType == newEncapsulationTypeValue
+
+        and: "Flow rules have been reinstalled"
+        !northbound.getSwitchRules(switchPair.src.dpId).flowEntries.findAll {
+            !new Cookie(it.cookie).serviceFlag
+        }.any { it in originalCookies }
+
+        cleanup: "Remove the flow"
+        flowHelperV2.deleteFlow(flow.flowId)
```

Comment:
'flow' may be null in case of assumption error

Reply:
fixed